classes/class-woothemes-sensei-certificate-data-cleaner.php

```diff
@@ -0,0 +1,294 @@
+ *
+ * @package Core
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	// Exit if accessed directly.
+	exit;
+}
+
+/**
+ * Class for cleaning up all plugin data.
+ *
+ * @author Automattic
+ * @since  2.2.2
+ */
+class WooThemes_Sensei_Certificate_Data_Cleaner {
+
+	/**
+	 * Custom post types to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $custom_post_types = array(
+		'certificate',
+		'certificate_template',
+	);
+
+	/**
+	 * Options to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $options = array(
+		'sensei_certificates_version',
+		'sensei_certificate_user_data_installer',
+		'sensei_certificate_templates_installer',
+		'sensei_certificates_view_certificate_button_added',
+	);
+
+	/**
+	 * Attachment GUIDs (as MySQL regexes) to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $attachment_guids = array(
+		'.+sensei_certificate_nograde.png$',
+	);
+
+	/**
+	 * Capabilities to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $caps = array(
+		// Certificates.
+		'edit_certificate',
+		'read_certificate',
+		'delete_certificate',
+		'create_certificates',
+		'edit_certificates',
+		'edit_others_certificates',
+		'publish_certificates',
+		'read_private_certificates',
+		'delete_certificates',
+		'delete_private_certificates',
+		'delete_published_certificates',
+		'delete_others_certificates',
+		'edit_private_certificates',
+		'edit_published_certificates',
+
+		// Certificate Templates.
+		'edit_certificate_template',
+		'read_certificate_template',
+		'delete_certificate_template',
+		'create_certificate_templates',
+		'edit_certificate_templates',
+		'edit_others_certificate_templates',
+		'publish_certificate_templates',
+		'read_private_certificate_templates',
+		'delete_certificate_templates',
+		'delete_private_certificate_templates',
+		'delete_published_certificate_templates',
+		'delete_others_certificate_templates',
+		'edit_private_certificate_templates',
+		'edit_published_certificate_templates',
+	);
+
+	/**
+	 * Transient names (as MySQL regexes) to be deleted. The prefixes
+	 * "_transient_" and "_transient_timeout_" will be prepended.
+	 *
+	 * @var string[]
+	 */
+	private static $transients = array(
+		'sensei_certificates_job_create_certificates',
+	);
+
+	/**
+	 * User meta key names (as MySQL regexes) to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $user_meta_keys = array(
+		'^%BLOG_PREFIX%sensei_certificates_view_by_public$',
+	);
+
+	/**
+	 * Post meta to be deleted.
+	 *
+	 * @var string[]
+	 */
+	private static $post_meta = array(
+		'_course_certificate_template',
+	);
+
+	/**
+	 * Cleanup all data.
+	 */
+	public static function cleanup_all() {
+		self::cleanup_custom_post_types();
+		self::cleanup_attachments();
+		self::cleanup_post_meta();
+		self::cleanup_capabilities();
+		self::cleanup_transients();
+		self::cleanup_settings();
+		self::cleanup_options();
+		self::cleanup_user_meta();
+
+		// Todo: Maybe cleanup the 'view-certificate' buttons.
```

Comment:
Does it make sense to also remove the view-certificate button block? The button block is added to the course completed page.

Reply:
#290 will add the button to the course page as well.

Reply:
Good point! I'm leaning towards leaving that up to the user to remove for now to avoid that extra complexity. We may need to reconsider if that other PR is merged.